bookshelf/api_v3/gce.py

```diff
@@ -0,0 +1,425 @@
+from time import time, sleep
+import uuid
+
+from zope.interface import implementer, provider
+from pyrsistent import PClass, field
+from oauth2client.client import (
+    SignedJwtAssertionCredentials, GoogleCredentials
+)
+from googleapiclient import discovery
+from googleapiclient.errors import HttpError
+
+from bookshelf.api_v2.logging_helpers import log_green, log_yellow, log_red
+from bookshelf.api_v1 import wait_for_ssh
+from cloud_instance import ICloudInstance, ICloudInstanceFactory, Distribution
+
+
+class GCEConfiguration(PClass):
```

Comment:
For the <cloud>Configuration and <cloud>State classes, would you like an explanation of each individual field (similar to what's in the CI-slave-images README)?